sys/kern/kern_sig.c

```diff
@@ -300,8 +300,9 @@ siginfo_to_siginfo_native(const _siginfo_t *si,
 	si_n->si_pid = si->si_pid;
 	si_n->si_uid = si->si_uid;
 	si_n->si_status = si->si_status;
-	si_n->si_addr = (void *)(uintptr_t)si->si_addr;
-	si_n->si_value.sival_ptr = (void *)(intcap_t)si->si_value.sival_ptr;
+	si_n->si_addr = (void *)(__cheri_addr vaddr_t)si->si_addr;
+	/* XXXAR: Do we want a CToPtr here? Is $ddc correct? */
+	si_n->si_value.sival_ptr = (__cheri_fromcap void *)si->si_value.sival_ptr;
```

Comment:
`si_value.sival_ptr` is a user-provided pointer-sized value; since this is the code for converting from a non-capability-aware user program, I think we want either an untagged capability whose cursor is the user `sival_ptr` or to fabricate a tagged capability using the user VA root capability in the kernel.  Since the kernel will not dereference this pointer, probably the former has better security properties.

Reply:
Why use the user's PCC rather than the root user VA cap?

Reply:
I don't think the user should be able to install a signal handler to something outside their current PCC (unless we add a mechanism for them to do so via a capability and install that as PCC when delivering the signal).  If you consider a world where you restrict legacy processes to a subset of the valid user VA range and install sandboxes (or other processes in the co-process world) outside of that region, then you don't want the legacy process to have the ability to manufacture access to things they can't otherwise reach via the kernel.

Reply:
Arg, too little tea this morning apparently.  You are right.  DDC should be used here.

Reply:
No worries.  The user's DDC seems like an OK choice to me, too. :)

Reply:
Sounds good, I'll split this out into a separate review. Other then this change okay to commit?